libfreerdp/core/message.c

```diff
@@ -107,10 +107,10 @@ static BOOL update_message_BitmapUpdate(rdpContext* context, BITMAP_UPDATE* bitm
 		wParam->rectangles[index].bitmapDataStream = (BYTE*) malloc(wParam->rectangles[index].bitmapLength);
 		if (!wParam->rectangles[index].bitmapDataStream)
 		{
-			for (index -= 1; index >= 0; --index)
-			{
-				free(wParam->rectangles[index].bitmapDataStream);
-			}
+			UINT32 i;
+			for (i = 0; i < index; ++i)
+				free(wParam->rectangles[i].bitmapDataStream);
+
 			free(wParam->rectangles);
```

Comment:
I don't think we are freeing the wrong pointers - only the ones that have been newly assigned to the BITMAP_DATA struct at line 107 (0 to index - 1).